features/other_b/skin_public.feature

```diff
@@ -111,11 +111,11 @@ Feature: Public skins
 
   Scenario: A user can preview a cached public site skin, and it will take the 
   user to the works page for a canonical tag with between 10 and 20 works
-  Given the approved public skin "Usable Skin"
-    And the skin "Usable Skin" is cached
-    And the canonical fandom "Dallas" with 2 works
+  Given the canonical fandom "Dallas" with 2 works
     And the canonical fandom "Major Crimes" with 11 works
     And the canonical fandom "Rizzoli and Isles" with 21 works
+    And the approved public skin "Usable Skin"
```

Comment:
The `canonical fandom X with Y works` steps use `FactoryGirl` to create the works, which means that they need `User.current_user` to be nil in order to work properly. But the `approved_public_skin` step involves logging in, which sets `User.current_user`.

(If you'd prefer, I can try adding `And I am logged out` instead of reordering the steps. I haven't tested it, but it seems like it ought to work if you think it'd be cleaner.)

Reply:
Yeah, I think we should leave these steps alone and add the log out step here instead:

https://github.com/otwcode/otwarchive/blob/fe59be31fb7ddd7c1e71742ea2f26e9bdb6781dd/features/step_definitions/skin_steps.rb#L75-L79

Some of the other skin steps have a similar problem but let's leave them be for now.